tests/contracts/solidity/RandomnessLotteryDemo.sol

```diff
@@ -0,0 +1,205 @@
+    bytes32 public SALT_PREFIX = "my_demo_salt_change_me";
+
+    /// @notice global number of request done
+    /// @dev This number is used as a salt to make it unique for each request
+    uint256 public globalRequestCount;
+
+    /// @notice The current request id
+    uint256 public requestId;
+
+    /// @notice The list of current participants
+    address[] public participants;
+
+    /// @notice The current amount of token at stake in the lottery
+    uint256 public jackpot;
+
+    /// @notice the owner of the contract
+    address owner;
+
+    constructor() RandomnessConsumer() {
+        owner = msg.sender;
+        globalRequestCount = 0;
+        requestId = 0;
+    }
+
+    function participate() external payable {
+        /// We retrieve the status of the request to know if it is
+        /// pending fulfillment
+        Randomness.Request memory request = randomness.getRequest(requestId);
+
+        /// If the request.id is 0, it means the request doesn't exist
+        if (request.id != 0) {
+            revert WaitingFulfillment();
+        }
+
+        //each player is compelled to add a certain ETH to join
+        if (msg.value != PARTICIPATION_FEE) {
+            revert InvalidParticipationFee(msg.value, MIN_FEE);
+        }
+        participants.push(msg.sender);
+        jackpot += msg.value;
+    }
+
+    function startLottery() external payable onlyOwner {
+        /// We retrieve the status of the request to know if it is
+        /// pending fulfillment
+        Randomness.Request memory request = randomness.getRequest(requestId);
+
+        /// If the request.id is 0, it means the request doesn't exist
+        if (request.id != 0) {
+            revert WaitingFulfillment();
+        }
+
+        if (participants.length < MIN_PARTICIPANTS) {
+            revert NotEnoughParticipants(participants.length, MIN_PARTICIPANTS);
+        }
+        if (participants.length >= MAX_PARTICIPANTS) {
+            revert TooManyParticipants(participants.length, MAX_PARTICIPANTS);
+        }
+
+        uint256 fee = msg.value;
+        if (fee < MIN_FEE) {
+            revert NotEnoughFee(fee, MIN_FEE);
+        }
+
+        /// Requesting NUM_WINNERS random words with a delay of DELAY_BLOCKS blocks
+        /// Refund after fulfillment will go back to the caller of this function
+        /// globalRequestCount is used as salt to be unique for each request
+        requestId = randomness.requestLocalVRFRandomWords(
+            msg.sender,
+            fee,
+            FULFILLMENT_GAS_LIMIT,
+            SALT_PREFIX ^ bytes32(globalRequestCount++),
+            NUM_WINNERS,
+            DELAY_BLOCKS
+        );
+    }
+
+    /// @notice Allows to increase the fee associated with the request
+    /// @dev This is needed if the gas price increase significantly before
+    /// @dev the request is fulfilled
+    function increaseRequestFee() external payable {
+        randomness.increaseRequestFee(requestId, msg.value);
```

Comment:
If requestId 0 is valid, then we should change this. I suggest:
- Initialize request_id to u256::max.
- Have a boolean in the SC indicating whether the lottery started. This function would check against this value.

My concern is, say request_id is valid and exists, but was made by another SC. In that case, if one calls `increaseRequestFee`() before calling `startLottery` it would increase the wrong request's fee.

Reply:
That is correct. I originally thought the requestId would be a hash so having a 0 value would not be expected.
I'm putting the u256::max